src/components/Details/VmDetails/VmDetails.js

```diff
@@ -0,0 +1,252 @@
+  const statusColor = props.on ? '#3f9c35' : '#a30000';
+
+  return (
+    <span>
+      <span style={{ color: `${statusColor}` }} className={statusIcon} /> {statusText}
+    </span>
+  );
+};
+
+OnOffReporter.propTypes = {
+  on: PropTypes.bool.isRequired,
+};
+
+const VmStatusReporter = props => {
+  const vmIsRunning = props.vm.spec.running;
+  const statusText = vmIsRunning ? 'on' : 'off';
+  const statusIcon = vmIsRunning ? 'pficon pficon-on' : 'pficon pficon-off';
+  const statusColor = vmIsRunning ? '#3f9c35' : '#a30000';
+
+  return (
+    <span style={{ color: `${statusColor}` }}>
+      <span className={statusIcon} /> {props.vm.metadata.name} is {statusText}
+    </span>
+  );
+};
+
+VmStatusReporter.propTypes = {
+  vm: PropTypes.object.isRequired,
+};
+
+export const VmDetails = props => {
+  const { loaded, NodeLink, resources, ResourceLink, vm } = props;
+  const pod = getPod(vm, resources, VIRT_LAUNCHER_POD_PREFIX);
+
+  // TODO Fix FieldLevelHelp content for Status
+  return (
+    <Fragment>
+      <div className="co-m-pane__body">
+        <h1 className="co-m-pane__heading">Overview</h1>
+        <div className="co-m-pane__body">
+          <div className="row">
+            {/* Description column */}
+            <div className="col-sm-4">
+              <div>
+                <VmStatusReporter vm={vm} />
+              </div>
+              <dd>{get(vm, 'metadata.annotations.description', DASHES)}</dd>
+            </div>
+
+            {/* Details columns */}
+            <div className="col-sm-8">
+              <div className="row">
+                {/* Details column 1 */}
+                <div className="col-sm-4">
+                  <dl>
+                    <dt>
+                      Status
+                      <FieldLevelHelp content={"This is the VM's status"} />
+                    </dt>
+                    <dd>
+                      <StateColumn vm={vm} loaded={props.loaded} resources={props.resources} />
+                    </dd>
+
+                    <dt>Operating System</dt>
+                    <dd>{get(vm, 'metadata.annotations["os.template.cnv.io"]', DASHES)}</dd>
+
+                    <dt>IP Addresses</dt>
+                    <dd>IP Address List</dd>
+
+                    <dt>Workload Profile</dt>
+                    <dd>{get(vm, 'metadata.annotations["workload.template.cnv.io"]', DASHES)}</dd>
+
+                    <dt>Template</dt>
+                    <dd>{get(vm, 'metadata.annotations["template.cnv.ui"]', DASHES)}</dd>
+                  </dl>
+                </div>
+
+                {/* Details column 2 */}
+                <div className="col-sm-4">
+                  <dt>FQDN</dt>
+                  <dd>{pod ? pod.spec.hostname : DASHES}</dd>
+
+                  <dt>Project</dt>
+                  <dd>Project Information</dd>
+
+                  <dt>Pod</dt>
+                  <dd>
+                    <FirehoseResourceLink
```

Comment:
So far we kept this retrieval/firehose-related logic in web-ui and just passed data objects down to the web-ui-components. I would prefer to use it even here. It's too specific for just the OKD internals.

This comment refers to functions around `getFlattenForKind` or `findVMI` too.